### ISSUE TITLE
Manual replaceOnChanges:["logGroupName"] required when log group name changed of pulumi resource

### ISSUE THREAD

**Issue by Jimmy89** [pulumi/pulumi-aws-native#823]
### What happened?

I tried to change an existing log group name of an AWS resource. When changing the name, the aws-native packages tries to update the log group name with a patch, which cannot be correct as you cannot change the name. Instead, the log group must be replaced. Similar issue as https://github.com/pulumi/pulumi-aws-native/issues/822

### Expected Behavior

Changing a logGroupName should be a replace action and be documented as such. 

### Steps to reproduce

```
new aws.logs.LogGroup(`log`, {
  logGroupName: "/aws/foo", // <-- change this after creation into "/aws/bar" or something
  retentionInDays: 7
}, { replaceOnChanges: ["logGroupName"] }); // <-- current workaround to prevent the error
```

### Output of `pulumi about`

```
Dependencies:
NAME                VERSION
@pulumi/aws-native  0.51.0
@pulumi/pulumi      3.55.0
@pulumi/random      4.11.2
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Jimmy89** [comment]
This should be added to the list of https://github.com/pulumi/pulumi-aws-native/issues/736

**Comment by danielrbradley** [comment]
Thanks @Jimmy89 I've added it to the tracking list in #736 👍

**Comment by mjeffryes** [comment]
This should be resolved by #1053; changes to logGroupName will result in a replace rather than an update.
